internal/backend/health/healthchecker/svc.go

```diff
@@ -0,0 +1,35 @@
+package healthchecker
+
+import (
+	"go.autokitteh.dev/autokitteh/internal/backend/db"
+	"go.autokitteh.dev/autokitteh/internal/backend/health/healthreporter"
+	"go.autokitteh.dev/autokitteh/internal/backend/temporalclient"
+	"go.uber.org/zap"
+)
+
+type healthChecker struct {
+	db     db.DB
+	z      *zap.Logger
+	tc     temporalclient.Client
+	checks map[string]healthreporter.HealthReporter
+}
+
+func New(db db.DB, z *zap.Logger, tc temporalclient.Client) healthreporter.HealthReporter {
+	checker := &healthChecker{db: db, z: z, tc: tc, checks: map[string]healthreporter.HealthReporter{}}
+
+	checker.checks["db"] = db
+	checker.checks["temporal"] = tc
+	return checker
+}
+
+func (h *healthChecker) Report() []error {
```

Comment:
Return a single error, use https://pkg.go.dev/errors#Join to join several.

Reply:
fixed